app/policies/tree_policy.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+# Policies for TreesController
+class TreePolicy < ApplicationPolicy
```

Comment:
Agree, I was thinking the same thing when I reviewed your GraphQL Git code. It's probably easier to just use the repository policy than having weird constructs when adding authorization to GraphQL.